src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -263,25 +251,34 @@ public AnalysisResults(bool ignoreCase, bool allAsciiIfIgnoreCase, int hashIndex
             public bool RightJustifiedSubstring => HashIndex < 0;
         }
 
+        private delegate ReadOnlySpan<char> GetSpan(string s, int index, int count);
```

Comment:
Gee I wish I could not have to use a `delegate` here, but at least building them as statics here will ensure we only build one bit of code that can be JITted into glory. If only we could do `Func<string, int, int, ReadOnlySpan<char>>` declares.